retry.go

```diff
@@ -0,0 +1,299 @@
+package relay
+
+import (
+	"log"
+	"sync"
+	"time"
+
+	"github.com/armon/relay/broker"
+)
+
+// retryBroker implements a resilient AMQP broker which automatically replaces
+// the underlying channel if an unexpected error is observed. Due to AMQP's
+// persistent connections, this becomes necessary to survive even minor network
+// hiccups which may temporarily sever the connection to the server.
+//
+// The retryBroker does not come without a price, however. Clients are still
+// responsible for deduplicating messages or handling them idempotently, as
+// duplications are a regular happening in the face of network errors due to
+// message redelivery provided by the protocol. It is also important to note
+// that acknowledgements - both ack and nack - cannot be retried across
+// connection resets. This stems from the fact that a given consumer's channel
+// cannot be revived and must be thrown away if the consumer is to be reused.
+type retryBroker struct {
```

Comment:
We take the same approach in the relayBroker - going to merge as-is, but easy enough to adjust if necessary.